geowebcache/diskquota/core/src/main/java/org/geowebcache/diskquota/QuotaStoreProvider.java

```diff
@@ -51,17 +51,14 @@ public void destroy() throws Exception {
 
     @Override
     public void afterPropertiesSet() throws Exception {
-        replaceH2WithHsql();
+        setHsql();
         reloadQuotaStore();
     }
 
-    private void replaceH2WithHsql() throws IOException, ConfigurationException {
-        // migrate existing H2 DB selection to HSQL DB
+    private void setHsql() throws IOException, ConfigurationException {
         DiskQuotaConfig config = loader.loadConfig();
-        if (config.getQuotaStore() != null && config.getQuotaStore().equals("H2")) {
-            config.setQuotaStore("HSQL");
-            loader.saveConfig(config);
-        }
+        config.setQuotaStore("HSQL");
```

Comment:
It used to be conditional behavior, it should not change anything if the dialect happens to be Oracle for example. I suggest just removing the method, there is nothing left to replace.